src/aabb.rs

```diff
@@ -348,6 +348,31 @@ impl<S: BaseFloat> Continuous<Aabb2<S>, Point2<S>> for Ray2<S> {
     }
 }
 
+impl<S: BaseFloat> Discrete<Aabb2<S>> for Ray2<S> {
+    fn intersects(&self, aabb: &Aabb2<S>) -> bool {
+        let ray = self;
+
+        let mut tmin = S::neg_infinity();
+        let mut tmax = S::infinity();
+
+        if ray.direction.x != S::zero() {
```

Comment:
would be best to get the direction having `max(abs(_))` instead of the first non-zero one, i.e. there will be issues if it's close to zero

Reply:
Hmm, ok, I just pulled the code from the Continuous implementation :)

Reply:
sorry about that, the continuous is far from perfect

Reply:
I can take a look at it tonight. I think the current version makes a tradeoff of performance vs accuracy. If you want to use it in a BVH for instance you don't want to traverse more than necessary, so there's some merit in that

Reply:
perhaps you misunderstood what I proposed. I meant selecting either `.x` or `.y` branch based on which absolute is higher. This is not really of any performance concern to compare 2 absolutes

Reply:
Hmm, not sure you can do that? The zero comparison is only there to avoid divide by zero for rays parallell to one of the axes.

Reply:
Actually, nvm, I think your code is fine :)